docs/celo-codebase/protocol/transactions/escrow.md

```diff
@@ -1,25 +1,224 @@
 ---
-title: Celo's Escrow Contract
-description: Introduction to the Celo Escrow contract and how to use it to withdraw, revoke, and reclaim funds.
+title: Escrow Payments
+description: Introduction to the Celo Escrow contract and how to use it to transfer, withdraw, and revoke funds.
 ---
-# Escrow 
+# Escrow payments
 
-Introduction to the Celo Escrow contract and how to use it to withdraw, revoke, and reclaim funds.
+Introduction to the Celo Escrow contract and how to use it to transfer, withdraw, and revoke funds.
 
 ___
 
-## What is the Escrow Contract?
+## Overview
 
-The `Escrow` contract utilizes Celo’s Lightweight identity feature to allow users to _send payments to other users who don’t yet have a public/private key pair or an address_. These payments are stored in this contract itself and can be either withdrawn by the intended recipient or reclaimed by the sender. This functionality supports _both_ versions of Celo’s lightweight identity: identifier-based \(such as a phone number to address mapping\) and privacy-based. This gives applications that intend to use this contract some flexibility in deciding which version of identity they prefer to use.
+The Escrow smart contract ([Escrow.sol](https://github.com/celo-org/celo-monorepo/blob/6b6ce69fde8f4868b54abd8dd267e5313c3ddedd/packages/protocol/contracts/identity/Escrow.sol)) is a [Core Contract](../../../learn/celo-stack#celo-core-contracts) on Celo that lets users make escrowed payments. In short, Alice can make a payment to Bob, before Bob has a Celo account. This is particularly useful when Alice wants to invite Bob to Celo by making an escrowed payment to him, which he can claim after creating an account.
 
-## How it works
+The contract provides two options for claiming an escrowed payment:
 
-If Alice wants to send a payment to Bob, who doesn’t yet have an associated address, she will send that payment to this `Escrow` contract and will also create a temporary public/private key pair. The associated temporary address will be referred to as the `paymentId`. Alice will then externally share the newly created temporary private key, also known as an _invitation_, to Bob, who will later use it to claim the payment. This paymentId will now be stored in this contract and will be mapped to relevant details related to this specific payment such as: the value of the payment, an optional identifier of the intended recipient, an optional amount of `attestations` the recipient must have before being able to withdraw the payment, an amount of time after which the payment will expire \(more on that in the “withdrawing” section below\), which asset is being transferred in this payment, etc.
+1. using the recipient's phone number as proof of identity, and
+2. using a secret private key as proof of identity.
 
-## Withdrawing
```

Comment:
We don't seem to have any details on how Bob can withdraw funds. We should definitely have some info about this (i.e. what is the message that they need to sign (with which keys) and pass into the withdraw function?). Ideally, we would include code snippets that show what functions Bob would call via ContractKit in order to withdraw funds. (This can be done more generally, since the process is the same after completing attestations (which should then be linked to), though up to you if you want to include notes "per-flow" on how to derive the PK needed to sign the message for withdrawing funds)